tests/unit/lms/services/lti_h_test.py

```diff
@@ -14,30 +15,68 @@ def test_sync_does_nothing_if_provisioning_is_disabled(
     ):
         ai_getter.provisioning_enabled.return_value = False
 
-        lti_h_svc.sync(groups=[h_group])
+        lti_h_svc.sync(h_groups=[h_group])
 
-        h_api.upsert_user.assert_not_called()
+        h_api.bulk_action.assert_not_called()
 
-    def test_sync_catches_HAPIErrors(self, h_api, lti_h_svc, h_group):
-        h_api.upsert_user.side_effect = HAPIError
+    def test_sync_catches_HAPIErrors(
+        self, h_api, lti_h_svc, h_group, group_info_service
+    ):
+        h_api.bulk_action.side_effect = HAPIError
 
         with pytest.raises(HTTPInternalServerError):
-            lti_h_svc.sync(groups=[h_group])
-
-
-class TestGroupUpdating:
-    def test_it_upserts_the_group(self, h_api, lti_h_svc, h_group):
-        lti_h_svc.sync([h_group])
+            lti_h_svc.sync(h_groups=[h_group])
 
-        h_api.upsert_group.assert_called_once_with(h_group)
-
-    def test_it_raises_if_upserting_the_group_fails(self, h_api, lti_h_svc, h_group):
-        h_api.upsert_group.side_effect = HAPIError("Oops")
-
-        with pytest.raises(HTTPInternalServerError, match="Oops"):
-            lti_h_svc.sync([h_group])
+        group_info_service.assert_not_called()
 
-    def test_it_upserts_the_GroupInfo_into_the_db(
+    def test_sync_calls_bulk_action_correctly(self, h_api, h_user, lti_h_svc):
+        groups = [
+            HGroup(
+                name=f"name_{i}", authority_provided_id="test_authority_provided_id",
+            )
+            for i in range(2)
+        ]
```

Comment:
```suggestion
        groups = factories.HGroup.create_batch(2)
```

Note that each group has a different unique `authority_provided_id`, those wouldn't be the same with real groups.

Reply:
This was written before the factory was in (the HGroup here was also originally not the same thing). I'll get it up to scratch.